ansys/mapdl/core/post.py

```diff
@@ -249,6 +264,468 @@ def freq(self) -> float:
+        Return the average element displacement for all components.
+
+        >>> arr = mapdl.post_processing.element_displacement("ALL")
+        >>> arr.shape
+        (2080, 3)
+        >>> arr
+        array([[ 1.07396154e-06, -9.03608033e-06, -5.17768108e-12],
+               [ 3.15631730e-06, -2.65527340e-05,  1.07714512e-11],
+               [ 5.12543515e-06, -4.31175194e-05,  2.19929719e-12],
+               ...,
+               [ 5.41204700e-06, -4.80335719e-05,  7.75819589e-11],
+               [ 3.33649806e-06, -2.96109417e-05,  1.44947535e-10],
+               [ 1.13836132e-06, -1.01038096e-05,  6.95566641e-11]])
+
+        """
+        check_elem_option(option)
+        component = component.upper()
+        check_comp(component, DISP_TYPE)
+
+        if component in ["ALL", "NORM"]:
+            disp = np.vstack((
+                self.element_values("U", "X", option),
+                self.element_values("U", "Y", option),
+                self.element_values("U", "Z", option)
+            )).T
+            if component == "NORM":
+                return np.linalg.norm(disp, axis=1)
+            return disp
+
+        return self.element_values("U", component, option)
+
+    def plot_element_displacement(
+        self, component="NORM", option="AVG", show_elem_numbering=False, **kwargs
+    ) -> CameraPosition:
+        """Plot element displacement.
+
+        Parameters
+        ----------
+        component : str, optional
+            Structural displacement component to retrieve.  Must be
+            ``"X"``, ``"Y"``, ``"Z"``, or ``"NORM"``.
+        option : str, optional
+            Option for storing element table data.  One of the
+            following:
+
+            * ``"MIN"`` : Store minimum element nodal value of the
+              specified item component.
+            * ``"MAX"`` : Store maximum element nodal value of the
+              specified item component.
+            * ``"AVG"`` : Store averaged element centroid value of the
+              specified item component (default).
+        show_element_numbering : bool, optional
+            Plot the element numbers of the elements.
+        **kwargs : dict, optional
+            Keyword arguments passed to :func:`general_plotter
+            <ansys.mapdl.core.plotting.general_plotter>`.
+
+        Returns
+        -------
+        pyvista.plotting.renderer.CameraPosition
+            Camera position from plotter.  Can be reused as an input
+            parameter to use the same camera position for future
+            plots.  Only returned when ``return_cpos`` is ``True``.
+
+        Examples
+        --------
+        Plot the mean normalized element displacement for the first
+        result in the "X" direction.
+
+        >>> mapdl.post1()
+        >>> mapdl.set(1, 1)
+        >>> mapdl.post_processing.plot_element_displacement(
+        ...     "NORM",
+        ...     option="AVG"
+        ... )
+
+        """
+        if component.upper() == "ALL":
```

Comment:
```suggestion
        if component.upper() not in ["X", "Y", "Z", "NORM"]: 
            raise ValueError(
                f'"component.upper()" not allowed in this context.  Select a '
                'single displacement component (e.g. "X" or "NORM")'
            )
```

Reply:
`"ALL"` is actually fine here since we're performing additional parameter checking `element_displacement` method.